Make November 2025 news public (#32005)
idaholab/moose: modules/doc/content/newsletter/2025/2025_11.md

## MOOSE Modules Changes

### Navier Stokes module

The Physics syntax for the weakly compressible Navier Stokes equations can now accept a `fp` parameter for the fluid properties
being used. This helps define the functor materials needed to compute fluid properties in the kernels forming the equations.

### Porous flow module

Two new `ActionComponents` were created to add injection and production wells in a compact syntax.
These wells are represented by multiple objects to represent the flow channel from the well, and the boundary condition
at the well.

### Heat transfer module

A linear finite volume implementation of the P1 radiation model was added, and can now be used for coupled convective flow problems alongside the linear finite volume implementation of Navier-Stokes.

### Subchannel module

A new [SCMClosure system](syntax/SCMClosures/index.md) was added to allow for user-developers to specify the closures to use
in the subchannel solve. For now, only the friction factors can be selected using this system. Developers may create new closures

> for friction factors, using the dedicated [SCMFrictionClosureBase.md] base class.

Author: ```suggestion
for friction factors, using the dedicated SCMFrictionClosureBase base class.
```

This file does not exist.

## MOOSE Modules Changes

### Navier Stokes module

The Physics syntax for the weakly compressible Navier Stokes equations can now accept a `fp` parameter for the fluid properties
being used. This helps define the functor materials needed to compute fluid properties in the kernels forming the equations.

### Porous flow module

Two new `ActionComponents` were created to add injection and production wells in a compact syntax.
These wells are represented by multiple objects to represent the flow channel from the well, and the boundary condition
at the well.

### Heat transfer module

The P1 radiation model was added with the linear finite volume discretization. The use of this discretization facilitates
coupling convective flow problems, solved notably in the Navier Stokes module using this discretization, directly alongside

> radiative heat transfer.

Reviewer: ```suggestion
A linear finite volume implementation of the P1 radiation model was added, and can now be used for coupled convective flow problems alongside the linear finite volume implementation of Navier-Stokes.
```

### Subchannel module

A new [SCMClosure system](syntax/SCMClosures/index.md) was added to allow for user-developers to specify the closures to use
in the subchannel solve. For now, only the friction factors can be selected in this system. Developers may create new closures

> in this system, with a dedicated base class for each physical quantity computed using a closure.

Reviewer: I think this could be made more clear. You're saying that I'd need two different base classes for computing friction factors vs. heat transfer coefficients?

> in this system, with a dedicated base class for each physical quantity computed using a closure.

Reviewer 2: ```suggestion
for friction factors, using the dedicated [SCMFrictionClosureBase.md] base class.
Other base classes will be created for the other each physical quantity computed using closures.
```

from reactor-module meshes to CSG used in Monte Carlo codes.

## libMesh-level Changes

## PETSc-level Changes

## Bug Fixes, Minor Changes, and Minor Enhancements

- Several controls that were unable to recover properly when set to execute on `TIMESTEP_END` have been fixed.
- The `simulation_time_interval` output parameter, which was not functioning, has been removed. The `min_simulation_time_interval`
  should be used instead.
- The distributed rectilinear mesh generator now only offers `edge`, `quad`, and `hex` element types, which
  were the only element types supported. Other element types can still be generated by conversion.
- The [SolutionUserObject.md] now caches retrieval calls at a given location, to prevent scaling with the square
  of the number of variables.
- The `Makefile` system in MOOSE now allows for linking additional objects directly to the executable using the `ADDITIONAL_EXEC_OBJECTS`
  variable.
- PETSc SNES options are no longer added by default when using the linear finite volume discretization, which does not use the nonlinear
  SNES solver.

> - Material property output for array material properties can now support arrays of more than 10 components.

Reviewer 2: maybe we should rename ArrayVariables as MultiComponentVaraibles but does not roll off the tongue either

> - Material property output for array material properties can now support arrays of more than 10 components.

Reviewer: Well array variables are perhaps most frequently represented as `RealEigenVector`, but conceptually yes we do use "array" in places for dynamically sized containers which are called `vector` in the `std`. But the name for array variables was born out of the fact that "vector" was already taken by `LAGRANGE_VEC/NEDELEC_ONE`. We probably should rename ArrayVariables.

Anyway for the sake of this line of documentation, what is the template argument to `MaterialProperty`?

> - Material property output for array material properties can now support arrays of more than 10 components.

Reviewer: Sorry @cticenhour somehow I missed that you had modified this line in the commit you pushed yesterday. I've further updated it a bit, so this is resolved now in my opinion
